embassy-stm32/src/f4/spi.rs

```diff
@@ -279,9 +282,24 @@ where
 
             for i in 0..(static_buf.len() - 1) {
                 let byte = static_buf[i];
-                nb::block!(write_sr(&spi, byte));
+                loop {
+                    let fut = InterruptFuture::new(&mut self.spi_int);
+                    match write_sr(&spi, byte) {
+                        Ok(()) => break,
+                        _ => {}
+                    }
+                    fut.await;
```

Comment:
I think if you create the `InterruptFuture` before writing the data, it will fire immediately, causing `fut.await` to return on the first poll, even if the write isn't done.

Reply:
you're correct; it's a txe interrupt.